### PR TITLE
Add the V (vector) extension to the riscv64-linux-android target spec

### PR DESCRIPTION
**Pull request by chriswailes**
This feature has been enabled and tested internally in the Android project.

### CODE REVIEW

**Comment by workingjubilee** [comment]
This is a tier 3 target, so arbitrary fiddling with its spec is permitted, and it does not encode extensions like `gc` into its name, so there is no question of whether it should be `riscv64gcv-linux-android`. And, yk, it's the target maintainer.

But I do think we should do one thing.

@chriswailes Can you mention the ISA requirements in the platform support file? You can use "currently... at least..." kind of cagey terminology if you want to avoid committing to stuff, but if someone out there tries this on a random RV64 board, even if they had previously gotten Android working on it, it's likely to fail now, and some boards claim they support the V extension but actually support a pre-ratification version only which is not binary-compatible with the finalized version. It'd be nice to not ask people to gaze into the rustc source to find out why.

**Comment by chriswailes** [comment]
Sure thing!  Thanks for the great suggestion.  I debated being a little tricky with the list and doing something like:

```
* a(tomics)
* d(ouble-precision floating-point)
```

but eventually decided against it.  Let me know if you'd like any changes or additions to the documentation.

**Comment by workingjubilee** [comment]
Thank you!

I know it's common for SBC makers and the like to carry a bunch of patches for a while until they get stuff upstreamed, so if anyone is already in the process of putting Rust and Android on RV64, hopefully this makes life easier for them.

@bors r+

File: compiler/rustc_target/src/spec/riscv64_linux_android.rs [hunk 1]
```diff
@@ -9,7 +9,7 @@ pub fn target() -> Target {
         options: TargetOptions {
             code_model: Some(CodeModel::Medium),
             cpu: "generic-rv64".into(),
-            features: "+m,+a,+f,+d,+c,+Zba,+Zbb,+Zbs".into(),
+            features: "+m,+a,+f,+d,+c,+Zba,+Zbb,+Zbs,+V".into(),
```
Comment (taiki-e) [inline compiler/rustc_target/src/spec/riscv64_linux_android.rs:12]: I think these need to be lower case.

```suggestion
            features: "+m,+a,+f,+d,+c,+zba,+zbb,+zbs,+v".into(),
```

https://github.com/rust-lang/rust/blob/130ff8cb6c3d62ed66daf652cbb5323d3f93c4fc/compiler/rustc_codegen_ssa/src/target_features.rs#L255-L257

The current one generates "recognized feature" warnings.

```
'+Zba' is not a recognized feature for this target (ignoring feature)
'+Zbb' is not a recognized feature for this target (ignoring feature)
'+Zbs' is not a recognized feature for this target (ignoring feature)
'+V' is not a recognized feature for this target (ignoring feature)
```

https://godbolt.org/z/cEq1d6vK6

**Comment by workingjubilee** [comment]
Eh, really?
@bors r-

File: compiler/rustc_target/src/spec/riscv64_linux_android.rs (hunk 1 above)
Comment (workingjubilee) [reply compiler/rustc_target/src/spec/riscv64_linux_android.rs:12]: ...Wow that's annoying.

Comment (chriswailes) [reply compiler/rustc_target/src/spec/riscv64_linux_android.rs]: Done

**Comment by workingjubilee** [comment]
rustc... really should have caught that on its own.

Anyway,
@bors r+

File: compiler/rustc_target/src/spec/riscv64_linux_android.rs (hunk 1 above)
Comment (workingjubilee) [reply compiler/rustc_target/src/spec/riscv64_linux_android.rs]: Thank you both.

**Pull request merged**